geps/gep-1713/index.md

```diff
@@ -969,3 +971,321 @@ Prior Discussions:
+	SupportGatewayListenerSet FeatureName = "GatewayListenerSet"
+```
+They will validate the following scenarios :
+
+1. AllowedListeners is not specified on the parent Gateway
+    - `Gateway.spec.allowedListeners` is not specified (defaults to None)
+
+    - The ListenerSet is not accepted with the following status :
+      | Type    | Status | Reason |
+      | -------- | ------- | ------- |
+      | Accepted  | False    | NotAllowed |
+
+    - The parent gateway has the following status :
+      | Type    | Status | Reason |
+      | -------- | ------- | ------- |
+      | AttachedListenerSets  | Unknown    | // any reason // |
+      | AttachedListeners  | 0    |  |
+
+    - The request on the ListenerSet port fails.
+
+1. ListenerSets are not allowed on the parent Gateway
+    - `Gateway.spec.allowedListeners.namespaces.from` is set to `None`
+
+    - The ListenerSet is not accepted with the following status :
+      | Type    | Status | Reason |
+      | -------- | ------- | ------- |
+      | Accepted  | False    | NotAllowed |
+
+    - The parent gateway has the following status :
+      | Type    | Status | Reason |
+      | -------- | ------- | ------- |
+      | AttachedListenerSets  | False    | ListenerSetsNotAllowed |
+      | AttachedListeners  | 0    |  |
+
+    - The request on the ListenerSet port fails.
+
+1. A listener on the ListenerSet has a protocol conflict with a listener on the Gateway
+    - The conflicting listener on the Gateway is accepted based on the [Listener Precedence](#listener-precedence) with the following status :
+      | Type    | Status | Reason |
+      | -------- | ------- | ------- |
+      | Conflicted  | False    | NoConflicts |
+
+    - The parent gateway has the following status :
+      | Type    | Status | Reason |
+      | -------- | ------- | ------- |
+      | AttachedListenerSets  | // depends on whether there are any other valid listeners in the attached ListenerSets // | ListenerSetsAttached |
+      | AttachedListeners  | // total number of child listenerSets // |  |
+
+    - The conflicting listener on the ListenerSet is not accepted with the following status :
+      | Type    | Status | Reason |
+      | -------- | ------- | ------- |
+      | Conflicted  | True    | ProtocolConflict |
+
+    - The ListenerSet has the following status :
+      | Type    | Status | Reason |
+      | -------- | ------- | ------- |
+      | Accepted  | True    | ListenersNotValid |
+
+    - The request to the conflicting listener on the Gateway succeeds.
+    - The request to the conflicting listener on the ListenerSet fails.
+
+1. A listener on a ListenerSet has a protocol conflict with a listener on another ListenerSet
+    - The conflicting listener on the older ListenerSet is accepted based on the [Listener Precedence](#listener-precedence) with the following status :
+      | Type    | Status | Reason |
+      | -------- | ------- | ------- |
+      | Conflicted  | False    | NoConflicts |
```

Comment:
Hmm, this means that the winning ListenerSet owner will have way to know that there is another ListenerSet that's in conflict with this one.

It feels a little better to me to have this ListenerSet have a `Conflicted: True`, with a reason of something like ConflictWinner (or some other better name), to indicate that there was a conflict, but it won. Then the losing ListnenerSet would get "ConflictLoser" or something (wow, that's a terrible Reason, but you get what I mean).

That way "Conflicted" means "there is a conflict with something", and there is a Reason to indicate what the result of the conflict was.

I suppose, though, that `Accepted` is probably also carrying the same information, so maybe those Reasons are redundant. Interested in other's thoughts here.